Support comma-delimited multi `@Param` or `@Header` (#4060)
line/armeria: site/src/pages/docs/server-annotated-service.mdx

</Tip>

> If the type is `List<?>` or `Set<?>`, multiple parameters exist with the same name in a query string can be
> injected, e.g. `/hello1?number=1&number=2&number=3`. Or you can use a comma-delimited parameter:
> `/hello1?number=1,2,3` You can use <type://@Default> annotation or `Optional<?>` class here, too.

Reviewer: 1) The sentence starts with `the type`, but it is unclear what type it is
2) The sentence starting with `Or` is not a complete sentence.

What do you think of the following?

```suggestion
If multiple parameters exist with the same name in a query string, they can be injected as a `List<?>`
or `Set<?>`, e.g. `/hello1?number=1&number=2&number=3`. If a single parameter is mapped to a
`List<?>` or `Set<?>`, it will be converted into comma-separated values, e.g. `/hello1?number=1,2,3`.
You can use <type://@Default> annotation or `Optional<?>` class here, too.
```

}
```

> If you specify query delimiter while building an annotated service, or a `Delimiter` annotation with a query

Reviewer 2: ```suggestion
If you specify query delimiter while building an annotated service, or a <type://@Delimiter> annotation with a query
```

Author: Just a question, what is difference between ``Delimiter`` and `<type://@Delimiter>`?

Reviewer 2: Good question. `<type://@Delimiter>` will link to the Javadoc like:
<img width="147" alt="image" src="https://user-images.githubusercontent.com/1866157/163784860-4e06af8e-f55c-4c7e-8104-de560ad68852.png">

query parameter, a single parameter will be converted into separated values if the parameter is mapped to a
`List<?>` or `Set<?>`, e.g. `/hello1?number=1,2,3`.

```java
sb.annotatedService()

>   .useQueryDelimiter(",")

Reviewer 2: ```suggestion
  .queryDelimiter(",")
```
